virtualmachine/vsphere/util.go

```diff
@@ -603,7 +669,22 @@ var createTemplateName = func(t string, ds string) string {
 var uploadTemplate = func(vm *VM, dcMo *mo.Datacenter, selectedDatastore string) error {
 	template := createTemplateName(vm.Template, selectedDatastore)
 	vm.datastore = selectedDatastore
+	basePath, err := ioutil.TempDir("", "")
+	if err != nil {
+		return err
+	}
+	defer func() {
+                if err := os.RemoveAll(basePath); err != nil {
+                        fmt.Errorf("can't remove temp directory, error: %s", err.Error())
+                }
+        }()
 	// Read the ovf file
+	if vm.OvaPathUrl != "" {
+                vm.OvfPath, err = downloadOva(basePath, vm.OvaPathUrl)
+                if err != nil {
+                        return err
+                }
+        }
```

Comment:
While reading the OvfPath for Ovf file, it will error out if the file doesn't exist. The value validation check is to be done in halo.